modules/fatfs/zephyr_fatfs_config.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#if FFCONF_DEF != 86631
+#error "Configuration version mismatch"
+#endif
+
+/*
+ * Overrides of FF_ options from ffconf.h
+ */
+#if defined(CONFIG_FS_FATFS_READ_ONLY)
+#undef FF_FS_READONLY
+#define FF_FS_READONLY	CONFIG_FS_FATFS_READ_ONLY
+#endif /* defined(CONFIG_FS_FATFS_READ_ONLY) */
+
+#if defined(CONFIG_FS_FATFS_MKFS)
+#undef FF_USE_MKFS
+#define FF_USE_MKFS	CONFIG_FS_FATFS_MKFS
+#else
+/* Note that by default the ffconf.h disables MKFS */
+#undef FF_USE_MKFS
+#define FF_USE_MKFS 1
+#endif /* defined(CONFIG_FS_FATFS_MKFS) */
+
+#if defined(CONFIG_FS_FATFS_CODEPAGE)
```

Comment:
I am not that familiar with the kconfig generator. Is it guaranteed, that CONFIG_FS_FATFS_CODEPAGE and the others will never be defined to 0?

Otherwise I would suggest to use
```suggestion
#if defined(CONFIG_FS_FATFS_CODEPAGE) && CONFIG_FS_FATFS_CODEPAGE
```

Or even betther
```suggestion
#if IS_ENABLED(CONFIG_FS_FATFS_CODEPAGE)
```
from the https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util_macro.h

Reply:
Kconfig lists allowed code pages; 0 is valid choice as long as it is allowed in code, Kconfig has no checks for that.
I can not use `IS_ENABLED`, because ffconf.h is included directly in FAT FS code, and using `IS_ENABLED` would mean that I directly included Zephyr headers and definitions into FAT FS module.
That is why I also redefine and undef `STRINGIFY`, instead of using Zephyr provided definition.